skyportal/handlers/api/listing.py

```diff
@@ -0,0 +1,181 @@
+import re
+
+from baselayer.app.access import auth_or_token
+from ..base import BaseHandler
+from ...models import (
+    DBSession,
+    User,
+    Obj,
+    Listing,
+)
+
+
+class UserObjListHandler(BaseHandler):
+    @auth_or_token
+    def get(self):
+        """
+        ---
+            description: get all objects corresponding to a specific user and that match a list name.
+            parameters:
+            - in: query
+              name: user_id
+              required: true
+              type: string
+            - in: query
+              name: list_name
+              required: false
+              type: string
+
+            description: |
+                find all objects saved to this list.
+                If not given will return all objects
+                saved by the user to all lists.
+
+            responses:
+              200:
+                content:
+                  schema:
+                    allOf:
+                      - $ref: '#/components/schemas/Success'
+                      - type: array
+                        items:
+                          description: array of listing objects
+        """
+
+        user_id = self.get_query_argument("user_id")
+
+        if User.query.get(user_id) is None:  # verify that user exists
+            return self.error(f'User "{user_id}" does not exist!')
+
+        user_id = int(user_id)
+
+        # verify that poster has read access to user_id's lists
+        if (
+            not self.associated_user_object.id == user_id
+            and "System admin" not in self.associated_user_object.permissions
+        ):
+            return self.error('Insufficient permissions to access this listing. ')
+
+        list_name = self.get_query_argument("list_name", None)
+
+        objects = DBSession().query(Listing).filter(Listing.user_id == user_id)
+        if list_name is not None and re.search(r'^\w+', list_name):
+            objects = objects.filter(Listing.list_name == list_name)
```

Comment:
Should fail if list name is invalid (i.e., there is no list in the db of that name).